project/Dependencies.scala

```diff
@@ -119,16 +133,17 @@ object Dependencies {
       val junit = "junit" % "junit" % junitVersion % "test" // Common Public License 1.0
       val logback = Compile.logback % "test" // EPL 1.0
 
-      val scalatest = "org.scalatest" %% "scalatest" % scalaTestVersion % "test" // ApacheV2
-      val scalacheck = "org.scalacheck" %% "scalacheck" % scalaCheckVersion % "test" // New BSD
+      // TODO DOTTY
+      val scalatest = DottyCompatModuleID("org.scalatest" %% "scalatest" % scalaTestVersion % "test").withDottyCompat(getVersion()) // ApacheV2
```

Comment:
All your problems relatd to scalatest stem from the fact that you're using the Scala 2 version of scalatest, so none of the macros are usable. You should drop the `withDottyCompat` for all dependencies which are published for Scala 3, this includes scalatest: https://repo1.maven.org/maven2/org/scalatest/scalatest_3.0.0-M3/

Reply:
Amazing didn't know about it! Thanks a lot @smarter !
Unfortunately, Scala 3 is not listed by Maven central:
https://mvnrepository.com/artifact/org.scalatest/scalatest

How do we know if a library is available or we should use the compatibility mode?

Reply:
scaladex can help: https://index.scala-lang.org/scalatest/scalatest/scalatest/3.2.3 (see the "version matrix" button on the right), or just browse https://repo1.maven.org/maven2 manually.

Reply:
Is Scala `3.0.0-M3` still missing?
And, found that, probably, this is a good starting point:
https://index.scala-lang.org/search?q=depends-on:lampepfl/scala3-library&page=1

Reply:
done in #29930